moon/controller_hauler_round2.py

```diff
@@ -58,13 +58,8 @@ def __init__(self, config, bus):
         self.rover_angle = None
 
     def run(self):
-
+        self.wait_for_init()
```

Comment:
if the code is terminated during this `wait_for_init()`  then it tries to run the main code anyway ... so maybe move it inside and not to try catch exception inside??